### DISCUSSION TITLE
Inline components vs full-page components? What's your preference?

### DISCUSSION THREAD

**Discussion by rolandorojas** [livewire/livewire#3656] [4. General]
I wanted to get your opinion on the use of inline components vs full-page components. 

**The case can be made that any page with some interactivity could be built as a full-page Livewire component**. You could also sprinkle some inline-components here and there in your regular blade file. 

## The case for making everything a full-page component:

- Pro: Great if most of your app has dynamic interfaces
- Pro: Less need to make multiple routes/controllers. It's easy to show, edit, and save data within one component vs going with REST routes.
- Con: Unnecessary for pages with no dynamic content. If we do traditional blade views for these, then the structure would be inconsistent with some pages using controllers and others full-page components. Not great IMO.

## The case for adding inline components

- Pro: Traditional route/controller structure is kept.
- Pro: Smaller components can be easily reused
- Con: Keeping components in sync when using the same set of records (think search box / table with results) requires the extra step of setting up events.

**What's your approach? Full-page components? Inline components? A mix?**

**Comment by joshhanley** [comment]
@rolandorojas I use full page components.

No longer use Laravel controllers at all 🙂 

But I also have inline components that I use inside my full page components, but I use them sparingly, as event based communication between components isn't ideal.

**Reply by oneloud91** [reply]
Are you still using full page components at current time @joshhanley  ? I saw some comment from laracasts say something like:
"The temptation is to end up with a god like livewire component. Of course a full page livewire component can still include child components but the temptation is to lump it all into one leading to large amounts of data passing back and forth on each update"

Does he mean full-page components might impact performance? or did i misunderstood something.

I like full-page components as well but still considering the performance impact it might cause if we use full-page components.

**Reply by joshhanley** [reply]
@oneloud91 yep still using full page components! 🙂 Any chance you can send the link so I can have a read?

Yes there is a risk of full page components containing a lot of data (but it comes down to how you build it). It just depends on the app and what you are trying to display on the screen.

For example, if you had a dashboard, which contains data from a few different models, if that was a single full page component, then sending data for that whole page could potentially impact performance.

I would still use a full page component in that instance (to keep it consistent across my app), but would also have child components, one each for the different sections on the dashboard. This has the added advantage that if you needed one area of the dashboard to be updated frequently, you could put a `wire:poll` just in that nested component, and only it refreshes instead of the whole page.

Hope this helps!

**Reply by oneloud91** [reply]
@joshhanley Thanks for the explanation. 

Thanks for the tips on nested component. I am also quite worry when it comes to dashboard since of course it need to a lot of data to be displayed. Though i might not need to update it frequently since so far my cases monitoring dashboard from time to time is not needed at the moment but in the future i may use the way you did as well. And for your reference, here is the link
[https://laracasts.com/discuss/channels/general-discussion/using-livewire-as-full-page-or-just-laravel](url)

It is in the last comment by Snapey.

**Reply by EriksonEnterprises** [reply]
I came here searching for if I should pull the trigger or not lol. But honestly even changing most of my controllers to full page Livewire components, I noticed a huge performance increase for some reason. From what I read, when you have full page components, it makes your requests less especially on initial load

Where before it had to load my controller & blade, then load my component(s). Now it just loads the main component directly.

Cutting out the middleman really helped my application already. Gonna just change all of them...

This is even with a resource heavy location checker that runs every 2 seconds on the dashboard 🙌

**Reply by rolandorojas** [reply]
It's been a while, but I tend to prefer full-page livewire components. It's just easier to share data among them.
